core/src/main/java/org/elasticsearch/action/fieldstats/FieldStats.java

```diff
@@ -449,7 +450,11 @@ public String getMaxValue() {
 
         @Override
         protected java.lang.Long valueOf(String value) {
-            return dateFormatter.parser().parseMillis(value);
+            try {
+                return dateFormatter.parser().parseMillis(value);
+            } catch (Exception e) {
```

Comment:
Can we support milliseconds through an explicit format parameter? We just got rid of this fallback mechanism in date fields for 2.0. It can be trappy when a user has no way to disable this builtin format.